### ISSUE TITLE
Worker threads for scanning

### ISSUE THREAD

**Issue by rkubik** [rkubik/scanr#4]
A single thread/process is not fast enough to scan large ranges of ports. We can use queues and worker threads to hand off jobs to threads. This should be invisible to the user by default but can be toggled by a command-line option.

```
# use 4 threads
scanr -t 4 10.10.10.10
```

**Comment by rkubik** [comment]
Possible design:

```
Main Thread
1. Read program input
2. Add ports to scan queue
3. Spawn X threads (based on input)
4. Pass threads program options (target, timeout)
5. Wait for threads to finish
6. Print results from finish queue

Port Scan Thread
1. Remove port from scan queue
2. Run port check
3. Run optional port analysis check
4. Add results to finish queue
5. If nothing left in scan queue, quit
```

Benchmark of scanr` as of commit ea4d4e4eb97423160d1e3f44484739207f83df71:

```
root# time scanr **** 1 3635
3632	CLOSED	0.05
3633	CLOSED	0.06
3634	CLOSED	0.06
3635	CLOSED	0.06
real	4m47.800s
user	0m0.100s
sys	0m0.704s
```

We can use this to benchmark our improvements with multi-threading.

**Comment by rkubik** [comment]
Fixed by #10.
